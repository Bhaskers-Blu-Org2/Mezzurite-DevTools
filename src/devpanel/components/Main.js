import React, { Component } from 'react';
import MezzuriteInfo from './MezzuriteInfo.js';

class Main extends Component {

    componentDidMount() {
        
    }

    componentWillUnMount() {

    }

    render() {
        return (
            <main className="Main">
                <MezzuriteInfo />
            </main>
        );
    }
}

export default Main;