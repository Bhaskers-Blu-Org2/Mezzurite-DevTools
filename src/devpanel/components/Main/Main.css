@media only screen {
  .main {
    flex: 1;
    overflow-y: auto;
  }

  .main--header {
    margin: 16px;
    text-align: center;
  }

  .main--header-center {
    align-items: center;
    display: flex;
    flex-direction: column;
    height: 100%;
    justify-content: center;
    line-height: 24px;
    margin: 0;
  }

  .main--header-link {
    color: var(--microsoft-blue);
    text-decoration: none;
  }

  .main--header-link:active {
    color: var(--microsoft-gray-darker);
  }

  .main--header-text {
    max-width: 80%;
    margin: 0;
    text-align: center;
  }
}

@media only screen and (min-width: 540px) {
  .main--header {
    text-align: start;
  }
}
