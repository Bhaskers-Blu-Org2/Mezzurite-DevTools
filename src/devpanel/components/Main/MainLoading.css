@media only screen {
  .capture-cycle--skeleton {
    padding-bottom: 4px;
  }

  .capture-cycle--skeleton-component {
    border-radius: 8px;
    height: 28px;
    margin: 0 16px 16px;
  }

  .capture-cycle--skeleton-title {
    height: 24px;
    border-radius: 8px;
    margin: 16px auto 36px;
    max-width: 50%;
  }

  .shimmer {
    background: var(--color-gray-darker);
    background-image: -webkit-gradient(linear, left top, right top, from(var(--color-gray-darker)), color-stop(20%, var(--color-gray-darkest)), color-stop(60%, var(--color-gray-darker)), to(var(--color-gray-darker)));
    background-image: -webkit-linear-gradient(left, var(--color-gray-darker) 0%, var(--color-gray-darkest) 20%, var(--color-gray-darker) 60%, var(--color-gray-darker) 100%);
    background-image: -o-linear-gradient(left, var(--color-gray-darker) 0%, var(--color-gray-darkest) 20%, var(--color-gray-darker) 60%, var(--color-gray-darker) 100%);
    background-image: linear-gradient(to right, var(--color-gray-darker) 0%, var(--color-gray-darkest) 20%, var(--color-gray-darker) 60%, var(--color-gray-darker) 100%);
    background-repeat: no-repeat;
    background-size: 200vw 200vw;
    opacity: 0.3;
    -webkit-animation-duration: 2s;
    -webkit-animation-fill-mode: forwards;
    -webkit-animation-iteration-count: infinite;
    -webkit-animation-name: placeholderShimmer;
    -webkit-animation-timing-function: linear;
  }
}

@media only screen and (min-width: 640px) {
  .capture-cycle--skeleton-title {
    margin: 16px 16px 36px;
    max-width: 30%;
  }
}

@-webkit-keyframes placeholderShimmer {
  0% {
    background-position: -100vw 0;
  }

  100% {
    background-position: 100vw 0;
  }
}
