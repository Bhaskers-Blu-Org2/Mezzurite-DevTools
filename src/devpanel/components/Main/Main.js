import React from 'react';
import { arrayOf, number, shape, string } from 'prop-types';

import './Main.css';
import CaptureCycle from '../CaptureCycle/CaptureCycle';

const Main = (props) => {
  const noActiveCaptureCycles = props.captureCycles != null && props.captureCycles.length === 0 && props.applicationLoadTime != null;

  return (
    props != null && <main className='main'>
      {props.applicationLoadTime == null && props.captureCycles == null &&
        <h2 className='main--header'>Mezzurite is working...</h2>
      }
      {props.applicationLoadTime != null && <h2 className='main--header'>App<span className='mobile-hidden'>lication</span> Load Time: {props.applicationLoadTime} ms</h2>}
      {props.captureCycles != null && props.captureCycles.length > 0 && props.captureCycles.map((captureCycle, captureCycleIndex) => {
        if (captureCycle != null) {
          return (
            <CaptureCycle
              captureCycleIndex={captureCycleIndex}
              key={`capture-cycle-${captureCycleIndex}`}
              timestamp={captureCycle.time}
              timings={captureCycle.componentTimings}
            />
          );
        }
      })}
      {noActiveCaptureCycles && <h3>No active capture cycles have completed.</h3>}
    </main>
  );
};

Main.propTypes = {
  applicationLoadTime: number,
  captureCycles: arrayOf(
    shape({
      componentTimings: arrayOf(shape({
        componentLoadTime: number,
        componentName: string
      })),
      time: string
    })
  )
};

export default Main;
