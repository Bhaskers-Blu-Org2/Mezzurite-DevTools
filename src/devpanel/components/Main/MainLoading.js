import React from 'react';

import './MainLoading.css';

const MainLoading = () => (
  <div className='capture-cycle--card capture-cycle--skeleton'>
    <div className='capture-cycle--skeleton-title shimmer' />
    <div className='capture-cycle--skeleton-container'>
      <div className='capture-cycle--skeleton-component shimmer' />
      <div className='capture-cycle--skeleton-component shimmer' />
      <div className='capture-cycle--skeleton-component shimmer' />
    </div>
  </div>
);

export default MainLoading;
