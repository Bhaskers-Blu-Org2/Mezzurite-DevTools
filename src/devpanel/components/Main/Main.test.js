import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Main from './Main';

describe('Main.js', () => {
  const renderer = new ShallowRenderer();

  it('should be loading when the props are null', () => {
    const tree = renderer
      .render(<Main />);
    expect(tree).toMatchSnapshot();
  });

  it('should render without captureCycles', () => {
    const tree = renderer
      .render(<Main applicationLoadTime={5.42} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render when captureCycles is an empty array', () => {
    const tree = renderer
      .render(<Main applicationLoadTime={5.42} captureCycles={[]} />);
    expect(tree).toMatchSnapshot();
  });

  it('should not error out when captureCycles contains a null element', () => {
    const tree = renderer
      .render(<Main applicationLoadTime={5.42} captureCycles={[ null ]} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render without applicationLoadTime', () => {
    const tree = renderer
      .render(<Main captureCycles={[
        {
          insideViewportComponents: [
            {
              componentLoadTime: 5.23,
              componentName: 'ComponentName'
            }
          ],
          time: 'time'
        }
      ]} />);
    expect(tree).toMatchSnapshot();
  });
});
