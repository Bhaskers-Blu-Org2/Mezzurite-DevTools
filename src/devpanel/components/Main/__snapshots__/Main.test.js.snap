// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main.js should be loading when the props are null 1`] = `
<main
  className="main"
>
  <h2
    className="main--header"
  >
    <span
      className="main--header-text"
    >
      Mezzurite was not detected. Learn more about Mezzurite 
      <a
        className="main--header-link"
        href="https://github.com/Microsoft/Mezzurite#onboarding"
        target="_blank"
      >
        on GitHub
      </a>
      .
    </span>
  </h2>
</main>
`;

exports[`Main.js should not error out when captureCycles contains a null element 1`] = `
<main
  className="main"
>
  <h2
    className="main--header"
  >
    App
    <span
      className="mobile-hidden"
    >
      lication
    </span>
     Load Time: 
    5.4
    ms
  </h2>
</main>
`;

exports[`Main.js should render when captureCycles is an empty array 1`] = `
<main
  className="main"
>
  <h2
    className="main--header"
  >
    App
    <span
      className="mobile-hidden"
    >
      lication
    </span>
     Load Time: 
    5.4
    ms
  </h2>
  <h3>
    No active capture cycles have completed.
  </h3>
</main>
`;

exports[`Main.js should render with loading as true 1`] = `
<main
  className="main"
>
  <h2
    className="main--header"
  >
    Mezzurite is working...
  </h2>
  <MainLoading />
</main>
`;

exports[`Main.js should render without applicationLoadTime 1`] = `
<main
  className="main"
>
  <h2
    className="main--header"
  >
    <span
      className="main--header-text"
    >
      Mezzurite was not detected. Learn more about Mezzurite 
      <a
        className="main--header-link"
        href="https://github.com/Microsoft/Mezzurite#onboarding"
        target="_blank"
      >
        on GitHub
      </a>
      .
    </span>
  </h2>
  <CaptureCycle
    captureCycleIndex={0}
    insideViewportComponents={
      Array [
        Object {
          "componentLoadTime": 5.23,
          "componentName": "ComponentName",
        },
      ]
    }
    timestamp="time"
  />
</main>
`;

exports[`Main.js should render without captureCycles and loading as false 1`] = `
<main
  className="main"
>
  <h2
    className="main--header"
  >
    App
    <span
      className="mobile-hidden"
    >
      lication
    </span>
     Load Time: 
    5.4
    ms
  </h2>
</main>
`;

exports[`Main.js should render without captureCycles and loading as true 1`] = `
<main
  className="main"
>
  <h2
    className="main--header"
  >
    App
    <span
      className="mobile-hidden"
    >
      lication
    </span>
     Load Time: 
    5.4
    ms
  </h2>
  <MainLoading />
</main>
`;
